lib/dictBuilder/zdict.c

```diff
@@ -968,6 +968,9 @@ static size_t ZDICT_addEntropyTablesFromBuffer_advanced(
 }
 
 /* Hidden declaration for dbio.c */
+#if defined(ZSTD_PROGRAMS_LINK_SHARED)
+ZSTDLIB_API
```

Comment:
Here, we could make `ZDICT_trainFromBuffer_unsafe_legacy()` function part of the "unstable"/"experimental" section of `zdict` API.

Another alternative is to not use it in `dibio.c` and use the safe variant instead.
It comes at an additional memory allocation cost, but might be worth it to remove this dependency.